pkg/volume/aws_ebs/aws_util_test.go

```diff
@@ -0,0 +1,232 @@
+				ZonesPresent:      true,
+				AvailabilityZones: "zoneX,zoneY",
+			},
+		},
+		// pass since nil node with allowedTopologies specified and DynamicProvisioningScheduling enabled
+		{
+			name: "Nil_Node_with_Multiple_Allowed_Topology_terms",
+			node: nil,
+			dynamicSchedulingEnabled: true,
+			allowedTopologies: []v1.TopologySelectorTerm{
+				{
+					MatchLabelExpressions: []v1.TopologySelectorLabelRequirement{
+						{
+							Key:    kubeletapis.LabelZoneFailureDomain,
+							Values: []string{"zoneX"},
+						},
+					},
+				},
+				{
+					MatchLabelExpressions: []v1.TopologySelectorLabelRequirement{
+						{
+							Key:    kubeletapis.LabelZoneFailureDomain,
+							Values: []string{"zoneY"},
+						},
+					},
+				},
+			},
+			reject: false,
+			expectedOptions: &aws.VolumeOptions{
+				ZonesPresent:      true,
+				AvailabilityZones: "zoneX,zoneY",
+			},
+		},
+		// pass since nil node with allowedTopologies specified and DynamicProvisioningScheduling enabled
+		{
+			name: "Nil_Node_with_Single_Allowed_Topology_term",
+			node: nil,
+			dynamicSchedulingEnabled: true,
+			allowedTopologies: []v1.TopologySelectorTerm{
+				{
+					MatchLabelExpressions: []v1.TopologySelectorLabelRequirement{
+						{
+							Key:    kubeletapis.LabelZoneFailureDomain,
+							Values: []string{"zoneX"},
+						},
+					},
+				},
+			},
+			reject: false,
+			expectedOptions: &aws.VolumeOptions{
+				ZonePresent:      true,
+				AvailabilityZone: "zoneX",
+			},
+		},
+		// fail since zones as well as allowedTopologies specified and DynamicProvisioningScheduling enabled
+		{
+			name:                     "Nil_Node_with_Single_Allowed_Topology_term",
+			node:                     nil,
+			storageParams:            map[string]string{"zones": "zoneX, zoneY"},
+			dynamicSchedulingEnabled: true,
+			allowedTopologies: []v1.TopologySelectorTerm{
+				{
+					MatchLabelExpressions: []v1.TopologySelectorLabelRequirement{
+						{
+							Key:    kubeletapis.LabelZoneFailureDomain,
+							Values: []string{"zoneX"},
+						},
+					},
+				},
+			},
+			reject: true,
+		},
+		// fail since multiple zones specified in single allowedTopologies term DynamicProvisioningScheduling enabled
+		{
+			name:                     "Nil_Node_with_Multiple_Zones_in_Allowed_Topology_term",
+			node:                     nil,
+			storageParams:            map[string]string{"zones": "zoneX, zoneY"},
+			dynamicSchedulingEnabled: true,
+			allowedTopologies: []v1.TopologySelectorTerm{
+				{
+					MatchLabelExpressions: []v1.TopologySelectorLabelRequirement{
+						{
+							Key:    kubeletapis.LabelZoneFailureDomain,
+							Values: []string{"zoneX"},
+						},
+						{
+							Key:    kubeletapis.LabelZoneFailureDomain,
+							Values: []string{"zoneY"},
+						},
+					},
+				},
+			},
+			reject: true,
+		},
+	}
+
+	for _, test := range tests {
+		if test.dynamicSchedulingEnabled {
+			utilfeature.DefaultFeatureGate.Set("DynamicProvisioningScheduling=true")
```

Comment:
you probably need to disable the feature gate in the else case, to handle the times when you call continue